packages/js/product-editor/src/components/product-mvp-feedback-modal/product-mvp-feedback-modal.tsx

```diff
@@ -18,28 +17,29 @@ import { Text } from '@woocommerce/experimental';
 import { __ } from '@wordpress/i18n';
 import { useDispatch } from '@wordpress/data';
 
+export type ProductMVPFeedbackModalProps = {
+	recordScoreCallback: (
+		checked: string[],
+		comments: string,
+		email: string
+	) => void;
+	onCloseModal?: () => void;
+	onSkipFeedback?: () => void;
+};
+
 /**
  * Provides a modal requesting customer feedback.
  *
- *
  * @param {Object}   props                     Component props.
  * @param {Function} props.recordScoreCallback Function to call when the results are sent.
  * @param {Function} props.onCloseModal        Function to call when user closes the modal by clicking the X.
  * @param {Function} props.onSkipFeedback      Function to call when user skips sending feedback.
  */
-function ProductMVPFeedbackModal( {
+export function ProductMVPFeedbackModal( {
```

Comment:
What's the reason for exporting types that are not used anywhere except for this file? I see you added `export` in multiple places like that (e.g. `FeedbackModalProps`).

I prefer exporting only when needed rather than just in case but it's personal preference so just asking about your opinion 🙌

Reply:
I always export the Props type definition with its Component event if the Props are never used elsewhere, specially if the component belongs to a library. Otherwise we disallow the ability to extend or inherit those Props when wrapping and enhancing the child component. And because of that we ended up duplicating types everywhere. I do agree with you about internal types, like for example the State type definition of a class component or any other type that is not needed elsewhere never.

Reply:
Makes sense, thanks!